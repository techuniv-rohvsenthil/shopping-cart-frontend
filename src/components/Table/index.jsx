import React from 'react';
import propTypes from 'prop-types';
import * as styles from './index.module.css';
import TableContent from '../TableContent';

const Table = (props) => {
    const { items } = props;
    const data = Object.keys(items).map(key => items[key] );
    if (!data.length) {
        return (
            <div className={styles.cardContainer}>
                <TableContent item="item" price="price" quantity="quantity" subtotal="subtotal" />
            </div>
        );
    }
    const rows = data.map((item) => (
        <TableContent item={item.name} price={item.price} quantity={item.quantity} subtotal={item.price*item.quantity} />
    ));
    rows.unshift(<TableContent item="item" price="price" quantity="quantity" subtotal="subtotal" />);
    return (
        <div >
            <div >
                {rows}
            </div>
        </div>
    );
};

Table.propTypes = {
    items: propTypes.objectOf(propTypes.object).isRequired,
};


export default Table;
