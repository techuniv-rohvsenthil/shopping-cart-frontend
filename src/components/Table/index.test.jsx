import React from 'react';
import { render } from '@testing-library/react';
import Table from './index';

const cart = [
    {
      item: "apple",
      price: 50,
      quantity: 2,
    },
    {
      item: "apple",
      price: 50,
      quantity: 2,
    },
    {
      item: "apple",
      price: 50,
      quantity: 2,
    },
    {
      item: "apple",
      price: 50,
      quantity: 2,
    },
  ]
  

describe('the Table component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Table items={cart}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});