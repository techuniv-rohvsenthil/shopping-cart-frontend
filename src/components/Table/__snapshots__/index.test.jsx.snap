// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Table component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        <div>
          item
        </div>
        <div>
          price
        </div>
        <div>
          quantity
        </div>
        <div>
          subtotal
        </div>
      </div>
      <div>
        <div />
        <div>
          50
        </div>
        <div>
          2
        </div>
        <div>
          100
        </div>
      </div>
      <div>
        <div />
        <div>
          50
        </div>
        <div>
          2
        </div>
        <div>
          100
        </div>
      </div>
      <div>
        <div />
        <div>
          50
        </div>
        <div>
          2
        </div>
        <div>
          100
        </div>
      </div>
      <div>
        <div />
        <div>
          50
        </div>
        <div>
          2
        </div>
        <div>
          100
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
